### PR TITLE
Fix thread filtering for subscriber wakeups

### LINKED ISSUE #89: Workloads process messages from all threads when THREAD_ID is missing/empty
## User request
When starting multiple threads, multiple workloads are created, but **each workload processes messages for all threads**. The expectation is that each workload only processes the thread it was created for.

Environment noted by reporter:
- latest bootstrap
- agent image: `ghcr.io/agynio/agent-init-agn:0.3.5`

### Expected behavior
Per `agynio/architecture`, workloads are scoped to a single `(agent_id, thread_id)` pair. The Orchestrator injects `THREAD_ID` into the workload env, and `agynd` should only pull/unack messages for that thread.

### Root cause
`agynd` currently treats `THREAD_ID` as optional:
- `internal/config/config.go` reads `THREAD_ID` but does **not** validate it.
- When `THREAD_ID` is empty, `internal/platform/threads.go` omits the thread filter in `GetUnackedMessages`, causing **all threads’ messages** to be returned and processed.

### Required changes
1) **Fail fast if `THREAD_ID` is missing/empty/invalid**
- Make `THREAD_ID` required and validate as UUID in config.
- Update call sites to use the validated UUID.
- Update config tests accordingly.

2) **Defense-in-depth: filter notification wakeups by thread**
- `internal/subscriber/subscriber.go` currently wakes on `message.created` events for any thread.
- Filter by `payload.thread_id` before waking when `THREAD_ID` is configured.

### Tests
- Config unit tests: missing/empty/invalid UUID should error; valid UUID should parse.
- Subscriber unit tests: wake only on matching `thread_id`.

References:
- Architecture intent: `agynio/architecture` (agents-orchestrator + agynd-cli docs)
- Investigation notes: `THREAD_ID` optional config + `threadFilter=nil` behavior when empty.

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- require THREAD_ID env values to be valid UUIDs and normalize them in config
- filter subscriber wakeups by payload thread_id for defense-in-depth
- add unit coverage for thread ID config validation and subscriber filtering

## Testing
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go test ./...`
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go vet ./...`

Closes #89

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- validate THREAD_ID env values as UUIDs and normalize in config
- filter subscriber wakeups by payload thread_id to only wake for the assigned thread
- added unit tests covering thread ID config validation and subscriber filtering

## Testing
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go test ./...`
  - Passed: 7 packages, Failed: 0, Skipped: 0
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go vet ./...`
  - Lint: no issues

File: internal/subscriber/subscriber_test.go [hunk 1]
```diff
@@ -1,10 +1,118 @@
...
+	go func() {
+		done <- sub.Run(ctx)
+	}()
+
+	responses <- makeMessageCreated(otherThreadID)
+
+	select {
+	case <-sub.Wake():
+		t.Fatal("unexpected wake signal")
+	case <-time.After(200 * time.Millisecond):
```
Comment (noa-lucent) [inline internal/subscriber/subscriber_test.go:164]: [minor] Consider adding direct unit tests for `payloadThreadID` edge cases — nil payload, nil fields, missing `thread_id` key, non-UUID string value. The function is currently only tested indirectly through the subscriber integration tests, and these branches are easy to cover with a small table-driven test.

**Review by noa-lucent — DISMISSED** [review]
Clean, well-scoped fix. Config now fails fast on missing/invalid THREAD_ID, and the subscriber correctly filters wakeups by thread. The `notificationsClient` interface extraction is a nice testability improvement.

One minor suggestion for additional edge-case coverage on `payloadThreadID` — see inline comment.

LGTM.

File: internal/subscriber/subscriber_test.go (hunk 1 above)
Comment (casey-brooks) [reply internal/subscriber/subscriber_test.go:164]: Added direct unit coverage for payloadThreadID edge cases (nil payload/fields, missing key, invalid value, valid normalization). Thanks for the suggestion!

**Comment by casey-brooks** [comment]
## Update
- added direct unit coverage for payloadThreadID edge cases (nil payload/fields, missing key, invalid value, valid normalization)

## Testing
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go test ./...`
  - Passed: 7 packages, Failed: 0, Skipped: 0
- `GOPRIVATE=github.com/agynio/* GONOSUMDB=github.com/agynio/* GONOPROXY=github.com/agynio/* go vet ./...`
  - Lint: no issues

**Review by noa-lucent — APPROVED** [review]
Previous feedback addressed — `TestPayloadThreadID` now covers all edge cases (nil payload, nil fields, missing key, empty/invalid value, and uppercase normalization). Clean table-driven test, consistent with the rest of the file.

No new issues found. LGTM.
